pkg/syncer/util.go

```diff
@@ -103,16 +103,23 @@ func IsValidVolume(ctx context.Context, volume v1.Volume, pod *v1.Pod, metadataS
 		log.Errorf("Error getting Persistent Volume for PVC %s in volume %s with err: %v", pvc.Name, volume.Name, err)
 		return false, nil, nil
 	}
-
-	// Verify if pv is vsphere csi volume
-	if (pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csitypes.Name) && (metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume == nil) {
-		log.Debugf("Pod %s does not have a valid vSphereVolume. Ignoring the pod update", pod.Name)
-		return false, nil, nil
-	}
-	//Verify if pv is vsphere volume and migration flag is disabled
-	if !metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume != nil {
-		log.Warnf("%s feature switch is disabled. Cannot update vSphere volume metadata %s for the pod %s", common.CSIMigration, pv.Name, pod.Name)
-		return false, nil, nil
+	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csitypes.Name {
```

Comment:
Is it more appropriate to split this `if` statement into two? The first would be if `Spec.CSI == nil` and we check for `Spec.vSphereVolume` and all the logic that's within this scope. 
Then secondly, check if `Spec.CSI.Driver != csitypes.Name` and just print and return false? Currently, even if the volume is owned by a different CSI driver, our debug logs are going to say "does not have a valid vSphereVolume", as opposed to "not created by vSphere CSI driver"

Reply:
+1
Done